core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -780,7 +780,9 @@ private void handleRequestOrResponseComplete() {
                         // Stop receiving new requests.
                         handledLastRequest = true;
                         if (unfinishedRequests.isEmpty()) {
-                            ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(CLOSE);
+                            ctx.executor().schedule(() ->
+                                            ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(CLOSE),
```

Comment:
Could we check if the channel is still active? Meanwhile, a channel may have been closed.

Reply:
Thanks for checking! I added to check `ctx.channel().isActive()` before executing the close task.